core/contractsdk/java/src/main/java/com/baidu/xuper/ContextImpl.java

```diff
@@ -99,6 +114,111 @@ public void deleteObject(byte[] key) {
         return;
     }
 
+//    @Override
+    public Contract.IteratorResponse newIterator(ByteString start, ByteString limit) {
+        Contract.IteratorRequest request = Contract.IteratorRequest.newBuilder().setHeader(this.header)
+                .setStart(start).setLimit(limit).build();
+        try{
+            Contract.IteratorResponse resp = this.client.newIterator(request);
+            return resp;
+        } catch (StatusRuntimeException e){
+            return null;
+        }
+    }
+
+    @Override
+    public Contract.Transaction queryTx(String txid) {
+        Contract.QueryTxRequest request = Contract.QueryTxRequest.newBuilder().setHeader(this.header)
+                .setTxid(txid).build();
+
+        try{
+            Contract.QueryTxResponse resp = this.client.queryTx(request);
+            return resp.getTx();
+        } catch (StatusRuntimeException e){
+            return null;
+        }
+    }
+
+    @Override
+    public Contract.Block queryBlock(String blockid) {
+        Contract.QueryBlockRequest request = Contract.QueryBlockRequest.newBuilder().setHeader(this.header)
+                .setBlockid(blockid).build();
+
+        try{
+            Contract.QueryBlockResponse resp = this.client.queryBlock(request);
+            return resp.getBlock();
+        } catch (StatusRuntimeException e){
+            return null;
+        }
+    }
+
+    @Override
+    public void transfer(String to, String amount) {
+        Contract.TransferRequest request = Contract.TransferRequest.newBuilder().setHeader(this.header)
+                .setTo(to).setAmount(amount).build();
+        this.client.transfer(request);
```

Comment:
Is transfer always success?

Reply:
It will throw an exception to the contract's caller when transfer failed.